src/complex/DataStructure/AbstractDataStore.hpp

```diff
@@ -166,9 +183,13 @@ class AbstractDataStore : public IDataStore
     using pointer = const T*;
     using reference = const T&;
 
-    ConstIterator() = delete;
+    ConstIterator()
+    : m_DataStore(nullptr)
+    , m_Index(0)
+    {
+    }
 
-    ConstIterator(const AbstractDataStore& dataStore, usize index)
+    ConstIterator(const AbstractDataStore* dataStore, usize index)
```

Comment:
```suggestion
    ConstIterator(const AbstractDataStore& dataStore, usize index)
```
Same as the non-const one.